pallets/asset/src/lib.rs

```diff
@@ -223,37 +206,55 @@ decl_storage! {
         #[deprecated]
         pub AssetMetadataNextGlobalKey get(fn asset_metadata_next_global_key): AssetMetadataGlobalKey;
 
-        /// A list of tickers that exempt all users from affirming the receivement of the asset.
-        pub TickersExemptFromAffirmation get(fn tickers_exempt_from_affirmation):
-            map hasher(blake2_128_concat) Ticker => bool;
+        /// A list of assets that exempt all users from affirming its receivement.
+        pub AssetsExemptFromAffirmation get(fn assets_exempt_from_affirmation):
+            map hasher(blake2_128_concat) AssetID => bool;
 
-        /// All tickers that don't need an affirmation to be received by an identity.
-        pub PreApprovedTicker get(fn pre_approved_tickers):
-            double_map hasher(identity) IdentityId, hasher(blake2_128_concat) Ticker => bool;
+        /// All assets that don't need an affirmation to be received by an identity.
+        pub PreApprovedAsset get(fn pre_approved_asset):
+            double_map hasher(identity) IdentityId, hasher(blake2_128_concat) AssetID => bool;
 
         /// The list of mandatory mediators for every ticker.
         pub MandatoryMediators get(fn mandatory_mediators):
-            map hasher(blake2_128_concat) Ticker => BoundedBTreeSet<IdentityId, T::MaxAssetMediators>;
-
-        /// Storage version.
-        StorageVersion get(fn storage_version) build(|_| Version::new(4)): Version;
+            map hasher(blake2_128_concat) AssetID => BoundedBTreeSet<IdentityId, T::MaxAssetMediators>;
 
-        /// The last [`AssetMetadataLocalKey`] used for [`Ticker`].
+        /// The last [`AssetMetadataLocalKey`] used for [`AssetID`].
         pub CurrentAssetMetadataLocalKey get(fn current_asset_metadata_local_key):
-            map hasher(blake2_128_concat) Ticker => Option<AssetMetadataLocalKey>;
+            map hasher(blake2_128_concat) AssetID => Option<AssetMetadataLocalKey>;
 
         /// The last [`AssetMetadataGlobalKey`] used for a global key.
         pub CurrentAssetMetadataGlobalKey get(fn current_asset_metadata_global_key): Option<AssetMetadataGlobalKey>;
+
+        /// All tickers owned by a user.
+        pub TickersOwnedByUser get(fn tickers_owned_by_user):
+            double_map hasher(identity) IdentityId, hasher(blake2_128_concat) Ticker => bool;
+
+        /// All security tokens owned by a user.
+        pub SecurityTokensOwnedByUser get(fn security_tokens_owned_by_user):
+            double_map hasher(identity) IdentityId, hasher(blake2_128_concat) AssetID => bool;
+
+        /// Maps all [`AssetID`] that are mapped to a [`Ticker`].
+        pub AssetIDTicker get(fn asset_id_ticker): map hasher(blake2_128_concat) AssetID => Option<Ticker>;
+
+        /// Maps all [`Ticker`] that are linked to an [`AssetID`].
+        pub TickerAssetID get(fn ticker_asset_id): map hasher(blake2_128_concat) Ticker => Option<AssetID>;
+
+        pub RngNonce get(fn rng_none): map hasher(identity) IdentityId => u64;
```

Comment:
I would recommend using the caller's account id, instead of their DID, since a company may have multiple signing services using different keys to create assets.
```suggestion
        pub AssetNonce: map hasher(identity) T::AccountId => u64;
```